python/lsst/ip/isr/crosstalk.py

```diff
@@ -659,139 +686,171 @@ def subtractCrosstalk(self, thisExposure, sourceExposure=None, crosstalkCoeffs=N
+                if badAmpDict[amp.getName()]:
+                    coeffs[index, :] = 0.0
+                    coeffs[:, index] = 0.0
+
+        # Compute backgrounds if requested.
+        backgrounds = {amp.getName(): 0.0 for amp in sourceDetector}
+        if backgroundMethod == "AMP":
+            backgrounds = {
+                amp.getName(): self.calculateBackground(
+                    sourceMaskedImage[self._getAppropriateBBox(
+                        amp,
+                        isTrimmed,
+                        False,
+                    )], badPixels)
+                for amp in sourceDetector
+            }
         elif backgroundMethod == "DETECTOR":
-            backgrounds = [self.calculateBackground(source, badPixels) for amp in sourceDetector]
+            background = self.calculateBackground(sourceMaskedImage, badPixels)
+            backgrounds = {amp.getName(): background for amp in sourceDetector}
 
-        crosstalkPlane = mask.addMaskPlane(crosstalkStr)
+        # Add the crosstalk mask plane to the target mask.
+        sourceCrosstalkPlane = sourceMaskedImage.mask.addMaskPlane(crosstalkStr)
+        if sourceExposure is not None:
+            targetCrosstalkPlane = targetMaskedImage.mask.addMaskPlane(crosstalkStr)
+        else:
+            targetCrosstalkPlane = sourceCrosstalkPlane
+
+        # If we are not doing subtrahend masking, the CROSSTALK mask bit
+        # is set according to the counts in the source pixel (above
+        # background), regardless of the crosstalk coefficient value.
         if not doSubtrahendMasking:
-            # Set the crosstalkStr bit for the bright pixels (those
-            # which will have significant crosstalk correction).  This
-            # mask will get copied to the other amplifiers as the
-            # crosstalk subtrahend is calculated.
+            # When we are not using subtrahend masking, we set the mask.
+            thresholdBackground = self.calculateBackground(sourceMaskedImage, badPixels)
             threshold = lsst.afw.detection.Threshold(minPixelToMask + thresholdBackground)
-            footprints = lsst.afw.detection.FootprintSet(source, threshold)
-            footprints.setMask(mask, crosstalkStr)
+            footprints = lsst.afw.detection.FootprintSet(sourceMaskedImage, threshold)
+            footprints.setMask(sourceMaskedImage.mask, crosstalkStr)
 
-        crosstalk = mask.getPlaneBitMask(crosstalkStr)
+        crosstalk = sourceMaskedImage.mask.getPlaneBitMask(crosstalkStr)
 
-        # Define a subtrahend image to contain all the scaled crosstalk signals
-        subtrahend = source.Factory(source.getBBox())
+        # Define a subtrahend image to contain all the scaled crosstalk
+        # signals. These will be applied to the target image.
+        subtrahend = targetMaskedImage.Factory(targetMaskedImage.getBBox())
         subtrahend.set((0, 0, 0))
 
-        coeffs = coeffs.transpose()
```

Comment:
I know the ticket mentions that things are pixel identical, so I'm guessing this transpose is no longer needed because of the naming swaps below?

Reply:
In some sense, the transpose was never needed because you could always change `[index1, index2]` to `[index2, index1]`.  But I think it may be that the `s` and `t` confusion led to the addition of this in the first place.